### PR TITLE
support `require SAN` clause

### PR DESCRIPTION
**Pull request by lysu**
### What problem does this PR solve? 

ref https://github.com/pingcap/tidb/issues/14759

### What is changed and how it works?

support `require SAN` in `create user`, `grant`, and so on

### Check List 

Tests 

 - Unit test

Code changes

 - Has exported function/method change
 - Has exported variable/fields change
 - Has interface methods change

Side effects

 - n/a

Related changes

 - Need to cherry-pick to the release branch(maybe 3.1)

---
This change is [Reviewable](https://reviewable.io/reviews/pingcap/parser/877)

### CODE REVIEW

File: misc.go [hunk 1]
```diff
@@ -219,6 +219,7 @@ var tokenMap = map[string]int{
 	"COMPRESSION":              compression,
 	"CONCURRENCY":              concurrency,
 	"CONFIG":                   config,
+	"SAN":                      san,
```
Comment (kennytm) [inline misc.go:222]: Please put this alphabetically.

File: parser.y [hunk 2]
```diff
@@ -323,6 +323,7 @@ import (
 	columnFormat          "COLUMN_FORMAT"
 	columns               "COLUMNS"
 	config                "CONFIG"
+	san                   "SAN"
```
Comment (kennytm) [inline parser.y:326]: ditto

**Comment by jackysp** [comment]
PTAL @kennytm

**Comment by kennytm** [comment]
Repairing CLA

**Pull request merged**